libdnf5/rpm/package.cpp

```diff
@@ -300,6 +301,58 @@ std::string Package::get_location() const {
     return libdnf5::utils::string::c_to_str(solvable_lookup_location(solvable, nullptr));
 }
 
+std::vector<std::string> Package::get_remote_locations(const std::set<std::string> & protocols) const {
+    // For installed and commandline packages return empty
+    Solvable * solvable = get_rpm_pool(base).id2solvable(id.id);
+    if (auto * repo = static_cast<repo::Repo *>(solvable->repo->appdata)) {
+        if (repo->get_type() == repo::Repo::Type::COMMANDLINE || repo->get_type() == repo::Repo::Type::SYSTEM) {
+            return {};
+        }
+    }
+
+    auto location = get_location();
+    if (location.find("://") != std::string::npos) {
+        return {location};
+    }
+
+    std::vector<std::string> remote_locations;
+
+    auto pkg_baseurl = get_baseurl();
```

Comment:
I am not sure why we test exclusively `get_baseurl();` and then `auto repo_baseurls = get_repo()->get_baseurls();`.

May I ask you for explanation?

Idditionally I think that the order is not perfect one. I would recommend to add first mirrors (from metalinks, mirrorlist) and the last add baseurls.

There is a problem in librepo with order of alternatives. We are unable to handle together baseurls and mirrorlists/metalinks. When baseurl is set it use only baseurl. I think `baseurl` should be used as a last option not the first one to split downloads to multiple providers.

Reply:
There are actually two baseurls:
- repo baseurl, configured in repo file. This is the base url for the whole repo (alternative repo source). You are correct about librepo I will switch the order of the loops.
- second is package baseurl, this can theoretically be different for each package. It looks like:
```xml
<package type="rpm">
  <name>CQRlib</name>
  <arch>src</arch>
  <version epoch="0" ver="1.1.1" rel="4.fc29"/>
  <checksum type="sha256" pkgid="YES">36eb2f6e63d3df64e61b14e570d35997a0f9357414ac1f6a637c48b971f968e5</checksum>
  <summary>ANSI C API for quaternion arithmetic and rotation</summary>
  <description>CQRlib is an ANSI C implementation of a utility library for quaternion
arithmetic and quaternion rotation math.</description>
  <url>http://cqrlib.sourceforge.net/</url>
  <size package="7780" installed="1628" archive="1888"/>
  <location xml:base="http://localhost:43223" href="src/CQRlib-1.1.1-4.fc29.src.rpm"/>    <-----------
  <format>
    <rpm:license>LGPLv2+</rpm:license>
    <rpm:vendor>Fedora Project</rpm:vendor>
    <rpm:group>System Environment/Libraries</rpm:group>
    <rpm:buildhost>77a43894e301</rpm:buildhost>
    <rpm:sourcerpm></rpm:sourcerpm>
    <rpm:header-range start="4504" end="6850"/>
    <rpm:provides>
      <rpm:entry name="CQRlib" flags="EQ" epoch="0" ver="1.1.1" rel="4.fc29"/>
    </rpm:provides>
  </format>
</package>
```

I was thinking that if a repo says the package is elsewhere it probably doesn't make sense to try to find it in the repo location. Therefore there is a separate check for package baseurl and if present it returns only the one location.